### ISSUE TITLE
XML reporter leaves empty file if the process is killed by a signal

### ISSUE THREAD

**Issue by horenmar** [philsquared/Catch#663]
According to the [reporter documentation](https://github.com/philsquared/Catch/blob/master/docs/command-line.md#choosing-a-reporter-to-use) the xml reporter is streaming and thus I would expect at least partial output if the process is killed a by SIGINT. This is true if I run the binary as `./a.out -r xml`, but if I run the binary as `./a.out -r xml -o outfile`, I end up with an empty file.

In contrast, if I use the default or `compact` reporter, the output on stdout and in the outfile will be the same, which is a partial output until the signal was encountered.

---

Using `JUnit` reporter seems to trigger an assertion:

```
a.out: ./catch.hpp:8734: virtual void Catch::CumulativeReporterBase::testCaseEnded(const Catch::TestCaseStats &): Assertion `m_sectionStack.size() == 0' failed.
```

**Comment by horenmar** [comment]
Strictly speaking this can be worked-around by redirecting std-out, but it should be investigated anyway.

**Comment by philsquared** [comment]
This is potentially fixed by 84af6bc955491d4b2a4c1d3c61e0b3516ce9f298 - but needs testing

**Comment by horenmar** [comment]
This seems to be fixed, just throwing in `std::endl`s is a bit of hack solution, so I am gonna tag this with revisit.
